src/state/multicall/hooks.ts

```diff
@@ -293,10 +296,10 @@ export function useSingleCallResult(
       : EMPTY_ARRAY
   }, [contract, fragment, inputs, gasRequired])
 
-  const result = useCallsData(calls, options)[0]
+  const { valid, data, blockNumber } = useCallsData(calls, options)[0] || {}
```

Comment:
Why do this?

Reply:
Because there are cases that `valid`, `data` and `blockNumber` doesn't had any change but they are wrapped in new object instance `result`. So watching `result` may cause unneccesary re-runs `useMemo`. Instead we only watch what we need here.
Note that `valid`, `data` and `blockNumber` are all primitive type.